### PR TITLE
test: migrate tests/k8s/tests/03-l7-stresstest.sh

### PR DESCRIPTION
**Pull request by ianvernon**
* Migrate bash-script test to Ginkgo framework.
* Add CreateResource and DeleteResource helper functions.

Signed-off by: Ian Vernon <ian@cilium.io>

Related-to: #1841

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

File: test/k8sT/Policies.go
Comment (aanm) [inline test/k8sT/Policies.go]: @eloycoto I've seen that you've written something like this in a previous PR and I've suggested to use `sync.Once`. 
@ianvernon @eloycoto Why don't we use `BeforeSuite` instead of `BeforeEach` for this test?

File: test/k8sT/Policies.go
Comment (aanm) [inline test/k8sT/Policies.go]: ~Yes, but leave it commented along with the comment it was on the previous test:~
```
  # FIXME: Due proxy constrains (memory?) it's impossible to execute the test
  # with 1000000 requests and 200 parallel connections. It was tested with
  # 1 request and 1 parallel connection with no success.
  #
  #kubectl exec -n qa -i ${frontend_pod} -- ab -r -n 1000000 -c 200 -s 60 -v 1 "http://${backend_svc_ip}:80/"
```

No, because this should be on the nightly tests

File: test/k8sT/Policies.go
Comment (aanm) [inline test/k8sT/Policies.go:390]: This doesn't work as you're expecting, check #2496

File: test/k8sT/Policies.go
Comment (aanm) [inline test/k8sT/Policies.go:404]: Where are those files?

File: test/k8sT/Policies.go
Comment (aanm) [inline test/k8sT/Policies.go]: Shouldn't this line be swapped with the next one? If there's an error, pods will be false and will we still know which error was it?

File: test/k8sT/Policies.go
Comment (aanm) [inline test/k8sT/Policies.go]: Out of curiosity, why are you printing this (from lines 426 to lines 436) again? It was printing on line 406

File: test/k8sT/Policies.go
Comment (aanm) [inline test/k8sT/Policies.go]: Unnecessary function?

File: test/k8sT/Policies.go
Comment (aanm) [inline test/k8sT/Policies.go]: you should also check in `ciliumPodK8s2`, no?

File: test/k8sT/Policies.go
Comment (aanm) [inline test/k8sT/Policies.go]: Unnecessary function as well?

File: test/k8sT/Policies.go
Comment (aanm) [inline test/k8sT/Policies.go]: Can you swap the `backend` with `frontend`? We usually test `from -> to`

File: test/k8sT/Policies.go
Comment (aanm) [inline test/k8sT/Policies.go]: `Testing cilium network policy enforcement from any namespace`

**Review by aanm — CHANGES_REQUESTED** [review]
Just to make sure, I assume you also saw this comment on the `03-l7-stresstest`, right?
```
# TODO: implement in nightly builds (GH-1503)
#log "running stress tests"
#kubectl exec -n qa -i ${frontend_pod} -- wrk -t20 -c1000 -d60 "http://${backend_svc_ip}:80/"
#kubectl exec -n qa -i ${frontend_pod} -- ab -r -n 1000000 -c 200 -s 60 -v 1 "http://${backend_svc_ip}:80/"
```

File: test/k8sT/Policies.go
Comment (eloycoto) [reply test/k8sT/Policies.go]: @aanm Yes, but I didn't migrate all the initialize. Sync.Once works perfect. 

The `beforeSuite` is used, but for creating all the needed vms.

Comment (aanm) [reply test/k8sT/Policies.go]: @eloycoto but can't we have multiple before suites?

Comment (ianvernon) [reply test/k8sT/Policies.go]: Good idea! I created an enhancement issue to track making this change across the tests :) https://github.com/cilium/cilium/issues/2500 

I'll add this as part of my next upload.

File: test/k8sT/Policies.go
Comment (ianvernon) [reply test/k8sT/Policies.go:390]: This directly is ported from the bash-script-based tests:
```
CILIUM_PODS=$(kubectl -n kube-system get pods -l k8s-app=cilium | grep cilium- | awk '{print $1}')

for pod in $CILIUM_PODS; do
    kubectl -n kube-system exec $pod -- cilium config Debug=false
done
```

File: test/k8sT/Policies.go
Comment (ianvernon) [reply test/k8sT/Policies.go:404]: *facepalm*, coming soon...

File: test/k8sT/Policies.go
Comment (ianvernon) [reply test/k8sT/Policies.go]: Not unnecessary; the defer call within it ensures that all policies are deleted within Cilium at the end of the function.

File: test/k8sT/Policies.go
Comment (ianvernon) [reply test/k8sT/Policies.go]: Fixed.

File: test/k8sT/Policies.go
Comment (ianvernon) [reply test/k8sT/Policies.go]: Fixed.

File: test/k8sT/Policies.go
Comment (ianvernon) [reply test/k8sT/Policies.go]: Comment removed.

File: test/k8sT/Policies.go
Comment (ianvernon) [reply test/k8sT/Policies.go:404]: Fixed!

File: test/k8sT/Policies.go
Comment (ianvernon) [reply test/k8sT/Policies.go]: Fixed.

File: test/k8sT/Policies.go
Comment (ianvernon) [reply test/k8sT/Policies.go]: Agreed, output was too verbose. Kept first instance of these.

File: test/k8sT/Policies.go
Comment (ianvernon) [reply test/k8sT/Policies.go]: Fixed.

File: test/k8sT/Policies.go
Comment (ianvernon) [reply test/k8sT/Policies.go]: Not unnecessary; the defer call within it ensures that all policies are deleted within Cilium at the end of the function.

File: test/k8sT/Policies.go
Comment (ianvernon) [reply test/k8sT/Policies.go]: Clarified.

**Comment by ianvernon** [comment]
test-me-please

File: test/k8sT/Policies.go
Comment (aanm) [reply test/k8sT/Policies.go:390]: Ok, when I wrote the test I thought it did one thing, I confirm it's no longer the case that's why I created issue #2496.

**Review by aanm — APPROVED** [review]

**Comment by ianvernon** [comment]
Bash test build failed for unknown reasons, so I'm going to re-trigger a build after rebasing.

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
Please hold off on merging this until #2516 is merged. I'd like to ensure this test runs reliably on Jenkins, as it is not actually ran as part of this job for the required CIlium-Bash-Tests and Cilium-Ginkgo-Tests builds.

**Comment by ianvernon** [comment]
test-all-ginkgo

**Comment by ianvernon** [comment]
Cilium-Ginkgo-Tests-All succeeded with this change. Thus, am merging.

**Pull request merged**
